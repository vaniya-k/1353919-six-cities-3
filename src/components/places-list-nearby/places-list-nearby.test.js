import React from 'react';
import renderer from 'react-test-renderer';
import PlacesListNearby from './places-list-nearby.jsx';
import placesListing from '../../mocks/places-listing-test.js';

it(`<PlacesListNearby/> should return 3 places displayed`, () => {
  const mock = jest.fn();

  const tree = renderer
    .create(<PlacesListNearby places={placesListing.places.splice(0, 3)} onPlaceCardClick={mock}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
