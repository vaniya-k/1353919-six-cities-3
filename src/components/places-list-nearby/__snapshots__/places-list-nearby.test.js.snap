// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlacesListNearby/> shows three places in Gotham 1`] = `
<section
  className="near-places places"
>
  <h2
    className="near-places__title"
  >
    Other places in the neighbourhood
  </h2>
  <div
    className="near-places__list places__list"
  >
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="near-places__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src="https://"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              456
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/place/3"
            onClick={[Function]}
          >
            Batman's Den
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          house
        </p>
      </div>
    </article>
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="near-places__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src="https://"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              789
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/place/4"
            onClick={[Function]}
          >
            Wayne Estate
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="near-places__card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="near-places__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src="https://"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              19
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/place/5"
            onClick={[Function]}
          >
            Circus
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          room
        </p>
      </div>
    </article>
  </div>
</section>
`;
